### ISSUE TITLE
[Bug] Prevent ETA model validation from crossing route and temporal boundaries

### ISSUE THREAD

**Issue by ThunderKhan** [KanishJebaMathewM/Truxify#15966]
**Describe the bug**

The ETA traffic pipeline currently builds the training dataset by concatenating sequence samples from multiple routes and then relies on Keras `validation_split=0.2`.

Because the sequence data is assembled route-by-route and then concatenated before the split, the validation set is not guaranteed to represent a clean temporal holdout for each route. This can place later observations from one route in training while earlier observations from the same route remain in validation, weakening the evaluation signal and allowing temporal leakage into model validation.

The validation set should represent future observations relative to the training observations for each route.

**To Reproduce**

1. Load historical traffic data containing multiple routes with timestamped observations.
2. Group the records by route and generate the 60-step LSTM sequences.
3. Concatenate the generated samples across routes.
4. Train the model using the existing `validation_split=0.2` behavior.
5. Inspect the resulting training and validation samples by route and timestamp.

The current implementation does not explicitly enforce a per-route chronological train/validation boundary.

**Expected behavior**

For every route, earlier observations should be used for training and the most recent observations should be reserved for validation.

The validation set should therefore be:
- route-specific;
- chronologically later than the corresponding training samples;
- deterministic across runs;
- free from cross-route sequence mixing.

**Screenshots**

Not applicable.

**Environment (please complete the following information):**
 - OS: Windows
 - App Version: Current development branch
 - Browser: Not applicable

**Additional context**

The fix introduces an explicit temporal holdout from each route before model fitting and passes the resulting validation arrays through `validation_data`.

The implementation also handles environments where TensorFlow is unavailable by keeping ETA model functionality disabled rather than failing during `TrafficPipeline` initialization.

Verified locally with:
`python -m pytest tests/test_eta_validation_split.py tests/test_traffic_pipeline.py -q`

Result: 18 passed.

Also verified with:
`python -m py_compile services/traffic_pipeline.py tests/test_eta_validation_split.py`

**Comment by ThunderKhan** [comment]
I’d like to claim this issue and take ownership of the fix.

I’ve implemented the correction on the `fix/eta-validation-split` branch, including the per-route temporal validation holdout and the TensorFlow-unavailable initialization handling. The targeted test suite is passing locally, and the changes are ready for review.

I’ll be opening the corresponding PR in a few minutes.
